internal/rbd/rbd_util.go

```diff
@@ -726,7 +726,7 @@ func (rv *rbdVolume) checkImageChainHasFeature(ctx context.Context, feature uint
 			if errors.Is(err, ErrImageNotFound) {
 				return false, nil
 			}
-			util.ErrorLog(ctx, "failed to get image info for %s: %s", vol, err)
+			util.ErrorLog(ctx, "failed to get image info for %s: %s", vol.String(), err)
```

Comment:
`.String()` gets called implicitly when using the `%s` formatting. I do not think this commit makes a difference?

Reply:
No, currently the log looks like below,
```text
21-04-30T23:21:52.581603843Z E0430 23:21:52.581535   24684 rbd_util.go:589] ID: 2896 Req-ID: 0001-0011-rook-ceph-0000000000000001-6fc62615-aa0a-11eb-b36c-0a580a800411 failed to get image info for {%!s(*[]util.TopologyConstrainedPool=<nil>) <nil> map[] csi-snap-bec4c10d-aa09-11eb-b36c-0a580a800411   172.30.41.228:6789,172.30.166.153:6789,172.30.55.186:6789  replicapool   csi-snap-bec4c10d-aa09-11eb-b36c-0a580a800411 %!s(rbd.FeatureSet=257)           %!s(int64=10737418240) %!s(bool=false) %!s(bool=false) %!s(bool=false) %!s(bool=false) <nil>  seconds:1619824736 nanos:697795370 %!s(*util.ClusterConnection=&{0xc0001a71e0 0xc000111620}) %!s(*rados.IOContext=&{0x7f0490078b20})}: image not found: RBD image not found
```
to fix this one am explicitly calling the String() function

Reply:
This is because the String function is implemented as a pointer method, let me change it to value type it should also fix the issue. sample code https://play.golang.org/p/2vCy5-a6B3m

Reply:
okay looks like am getting a CI error if I consider pass by value
```
hugeParam: ri is heavy (208 bytes); consider passing it by pointer (gocritic)
```

Reply:
Hmm, maybe it is needed to pass `&vol` to the log function to get the `.String()` automatic?
See https://play.golang.org/p/VlyKSYbO3uv for my modified test.

I do not really care that much though, using `.String()` is fine, as we have that in many other places already.

Reply:
See #2051 for an issue that can be used to cleanup the `.String()` calling.